src/components/Pagination/index.tsx

```diff
@@ -0,0 +1,474 @@
+import styled from '@emotion/styled'
+import PropTypes from 'prop-types'
+import React, {
+  ForwardRefExoticComponent,
+  FunctionComponent,
+  ReactNode,
+  RefAttributes,
+  createContext,
+  forwardRef,
+  useCallback,
+  useContext,
+  useEffect,
+  useImperativeHandle,
+  useMemo,
+  useRef,
+  useState,
+} from 'react'
+import ActivityIndicator from '../ActivityIndicator'
+import Box from '../Box'
+import Button from '../Button'
+import getPageNumbers from './getPageNumbers'
+import usePagination, { UsePaginationReturn } from './usePagination'
+
+type PaginationContextValue = {
+  canLoadMore: boolean
+  data?: unknown[] | null
+} & UsePaginationReturn<unknown>
+
+export type PaginationState<T = unknown> = {
+  canLoadMore: boolean
+  data?: T[] | null
+} & UsePaginationReturn<T>
+
+const PaginationContext = createContext({} as PaginationContextValue)
```

Comment:
```suggestion
const PaginationContext = createContext<PaginationContextValue>({})
```

Maybe ?

Reply:
Doesn't work need to cast the object to the desired type :/